lib/v2/luogu/userBlog.js

```diff
@@ -0,0 +1,44 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
+
+module.exports = async (ctx) => {
+    const username = ctx.params.username;
+    const rsp = await got.get(`https://www.luogu.com.cn/blog/${username}/_sitemap`);
+    const $ = cheerio.load(rsp.data);
+    const blogTitle = $('title').text();
+    const blogLink = $('.am-alert > a').attr('href');
+
+    const metas = $('.am-u-md-12 .lg-table-big')
+        .map(function () {
+            const a = $(this).find('a').first();
+            return {
+                title: a.text(),
+                link: a.attr('href'),
+            };
+        })
+        .get();
```

Comment:
You can use the API endpoint instead
API unofficial docs: https://0f-0b.github.io/luogu-api-docs/
1. https://www.luogu.com.cn/api/user/search?keyword=${username}
2. https://www.luogu.com.cn/api/blog/lists?uid=${uid}&keyword=&type=&page=1

Reply:
Thanks for the doc. I will adopt it later.

Reply:
The full-text should be fetched from the API too. I will push it later.

Reply:
Maybe not. The `GET /api/blog/detail/:id` API doesn't seem to work.

Reply:
FYI, the blog detail API redirects me to its login page